File: tests/benchmark/vulnerable/javascript/xss_001.js
// BENCHMARK: vulnerable - xss
const express = require('express');
const app = express();

app.get('/search', (req, res) => {
    const query = req.query.q;
    res.send(`<div id="result"></div>
<script>
  document.getElementById('result').innerHTML = '${query}';
</script>`);
});

app.listen(3000);

File: tests/benchmark/vulnerable/python/xss_001.py
# BENCHMARK: vulnerable - xss
from flask import Flask, request
from markupsafe import Markup

app = Flask(__name__)


@app.route("/comment")
def show_comment():
    comment = request.args.get("comment", "")
    safe_comment = Markup(comment)
    return f"<div>{safe_comment}</div>"
